Fixes #1414 Decreasing the size of img folder by 55 % ( using webp instead of png , jpg ) and decreasing the start-up time

subhahu123 opened the pull request:
<!-- Add issue number here. If you do not solve the issue entirely, please change the message e.g. "Addresses #IssueNumber -->
Fixes #1414 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link
<!-- Replace "PR_NUMBER" with your pull request number. This link is generated when your PR passes the travis tests.A sample link can look like https://pr-200-fossasia-susper.surge.sh -->

#### Changes proposed in this pull request:

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
- using webp image format ( created by google ) instead of png, jpg,ico

<!-- Demo Link: Add here the link where you changes can be seen. -->

- https://pr-1413-fossasia-susper.surge.sh

<!-- Screenshots for the change: Add here the screenshot of the fix. -->
- initially img folder size was around 1.5mb but total sizes of webp images is around 700kb.

subhahu123 committed:
using webp image instead of jgp,png

subhahu123 commented:
@praveenojha33  @shreyanshdwivedi  @rajvaibhavdubey 
please review this

rajvaibhavdubey reviewed (approved):
@subhahu123 Check compatibility on firefox
![screenshot from 2019-01-16 22-11-36](https://user-images.githubusercontent.com/33062425/51264138-c844e000-19db-11e9-8817-13c3bc4884ac.png)
else LGTM :+1: 
[reference](https://havecamerawilltravel.com/photographer/webp-website/)

rajvaibhavdubey reviewed (changes requested):
@subhahu123 .webp is not compatible on firefox

subhahu123 commented:
@rajvaibhavdubey  i will add the fallback images in next pr

praveenojha33 reviewed (changes requested):
Changes are not working in Firefox !!

subhahu123 committed:
Update 404.html

subhahu123 commented:
@praveenojha33  I have the requested changes 
Please review

praveenojha33 reviewed (changes requested):
Please fix conflicts. @AakashMallik @shreyanshdwivedi Please review.

subhahu123 commented:
@praveenojha33  sorry but what conflicts
there are no conflicts in this

shreyanshdwivedi reviewed (commented):
@subhahu123 Please squash your commits. The conflicts about which @praveenojha33 is talking about can be seen by you once you update your development branch and then rebase your branch (having this PR) with the dev branch. Please fix the conflicts and force push once everything is up to date :)

subhahu123 commented:
@praveenojha33  done with conflicts

praveenojha33 commented:
It still has conflicts(I think this is because of recent merged commits). Try updating your branch again.

subhahu123 committed:
adding webp image 

using webp images instead of jpg,png
Update search-bar.component.html
Update results.component.html
Update help.component.html
Update dropdown.component.html
Update contact.component.html
Update styles.css
Update styles.css
Fix #1388 : did you mean content is now clickable

Adding rel="noopener" for fixing cross-link links
Fixing not found result page
Update results.component.ts
Update results.component.ts
Update results.component.html
Update results.component.html
Refactors Theme Component and added tests for coverage

Merge branch 'development' into patch-17
Fixes #1327: Contact componnet refactored (#1328)


Fixes  #1317: Reducers and Actions cleaned (#1318)


Update contact.component.html
Merge branch 'development' into patch-17
Update dropdown.component.html
Update dropdown.component.html
Update help.component.html
Update results.component.html
Update results.component.html
Update search-bar.component.html
Update search-bar.component.html
Update results.component.html
Update help.component.html
Update dropdown.component.html

subhahu123 commented:
@praveenojha33  please try now

sakshee-19 commented on src/app/dropdown/dropdown.component.html:
```diff
@@ -23,17 +26,19 @@
-              <div class="img"><img src="../../assets/images/bug.png"></div>
+              <div class="img"><img src="../../assets/images/bug.webp" onerror="this.onerror=null; this.src='../../assets/images/bug.png'"></div>
+
+
```
Remove extra lines.

sakshee-19 reviewed (commented):
Just a minor change.

Commit:
Merge branch 'development' of https://github.com/fossasia/susper.com into fossasia-development

subhahu123 committed:
Added YouTube link in the infobox (#1247)


Fixed clear search issue #1230

Fixed double quotes to single quotes

Fixed reference for navbar html

Fixed spacing issues in constructor scope

Fixed clear search issue #1230

Add Flex Slider to Help Videos (#1234)

Switch

Lint
(#1207)Added link to map address

Analytics border lie inside the window (#1346)


Refactors Theme Component and added tests for coverage

Fixed Contact Form UI (#1257)

Fix 1179 - Add SUSI.AI for Magic Mirror, Yaydoc, Meilix and Phimp.me in Projects section under the menus icon in the navbar

Fixed UI of Advanced Search Page

add a share button to infobox

add a share button to infobox

fix indentation

add modal

add icons

fix linting errors

fix icons

fix travis build errors

fix whitespace

fix whitespace

format HTML

add borders

improve styles

increase padding

Style and link fixes on the Privacy and Terms page(#1261)

Tests Added (#1278)

* Tests Added

Some of the tests for the files have been added

* Tests Added

Some of the tests for the newly added functions have been added

Fixes #1327: Contact componnet refactored (#1328)


Fixes #1267: Infobox share modal, styling fixes, redundancies removed

fix footer

fixing (#1351)

Update contact.component.ts
Update contact.component.html
Fixed validations in Contact form

Fixed linting issue

Added share link and made styles changes to infobox-share-popup (#1349)


Fixes  #1317: Reducers and Actions cleaned (#1318)


Fixed formatting of code

Fixed a typo in the contact form

Refactored Help Component and Added tests for increasing coverage

New Design of about component

The design of the about component has been changed

Fixes #1221

Susper logo Addition
cursor chnages to hover

Update README.md
Fixed minor link and ui issues

depth field only accepts numbers

Loading animation added to result component (#1204)


Fixes #1308: Refactor app.modules.ts (#1309)


Fixing the susper logo alignment and its container
Fixing alignment for all screen sizes
Fixing width of div
Update index.component.html
Update index.component.css
Update index.component.html
Update index.component.css
Update index.component.css
Update index.component.css
Update index.component.css
Fixse #1306: Removed redundant package delarations (#1307)


Fixing About page css (#1298)


Added sidebar menu on Terms page and tests to increase coverage

Fixing the hovering and visited color of links (#1293)

* Changing color of hovering , Visited link 

Ignoring the overriden of tags of hover and visited

* Changing color

Fix 1268 - Fix overlapping Help Section with footer (#1271)

* Updated Badgeyay's icon (#1203)

The badgeyay's icon on susper.com has been updated.

Fixes #1201

* "Commit "

Added sidebar for Privacy page and tests to increase coverage

Fixes #1242 corrections in crawl job page (#1243)


Fixes #1310: Travis issue fixed

Fix #1388 : did you mean content is now clickable

Added/Updated tests

Fixes #1322: About component refactored and cleaned up

Adding rel="noopener" for fixing cross-link links
Update dropdown.css

To hide overflow along x-axis in the dropdown to give UI effect similar to that of Google Search
Update about.component.html
Update about.component.html
Update about.component.html
Update about.component.html
Update about.component.html
Update about.component.html
Update about.component.html
Update about.component.html
Update about.component.html
Update about.component.html
Update about.component.html
Adding a lang attribute to the html element
refactored

type changed

Search bar is fixed when scrolled

Removed fixed navbar for mobile

Changed shape of search-bar to curve shape (#1462)

* Changed shape of search-bar to curve shape

* Improved the corner of search-bar

Merge branch 'development' of https://github.com/fossasia/susper.com into fossasia-development

Merge branch 'fossasia-development' into development

Conflicts:

Commit:
Merge branch 'fossasia-development' into development

subhahu123 committed:
Delete 16x9.webp

subhahu123 committed:
Delete JsonGreen.webp

subhahu123 committed:
Delete Pslab.webp

subhahu123 committed:
Delete RssOrange.webp

subhahu123 committed:
Delete YaCyLogo2011_60.webp

subhahu123 committed:
Delete yaydoc.webp

subhahu123 committed:
Delete terminal.webp

subhahu123 committed:
Delete susper.webp

subhahu123 committed:
Delete badgeyay.webp

subhahu123 committed:
Delete blog.webp

subhahu123 committed:
Delete boat.webp

subhahu123 committed:
Delete bug.webp

subhahu123 committed:
Delete code.webp

subhahu123 committed:
Delete susimagicmirror.webp

subhahu123 committed:
Delete susi_60x12.webp

subhahu123 committed:
Delete search.webp

subhahu123 committed:
Delete phimp.webp

subhahu123 committed:
Delete not-found-bg.webp

subhahu123 committed:
Delete mountain.webp

subhahu123 committed:
Delete microphone.webp

subhahu123 committed:
Delete meilix.webp

subhahu123 committed:
Delete loklak.webp

subhahu123 committed:
Delete loading.webp

subhahu123 committed:
Delete fossasia.webp

subhahu123 committed:
Delete external.webp

subhahu123 committed:
Delete eventyay.webp

subhahu123 committed:
Delete contact.webp

Commit:
Merge branch 'development' of https://github.com/subhahu123/susper.com into development

Commit:
Merge remote-tracking branch 'upstream/development' into development

subhahu123 commented:
@praveenojha33  @simsausaurabh  
please merge this
done with conflicts and rebasing
